lemminx-maven/src/test/java/org/eclipse/lemminx/extensions/maven/InitMavenRequestTest.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Red Hat Inc. and others.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.lemminx.extensions.maven;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.util.Properties;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(NoMavenCentralIndexExtension.class)
+public class InitMavenRequestTest {
+
+	private static final String TEST_LOCAL_REPO_PATH = "/.m2/bug383repository";
+	private static final String TEST_DUMMY_PROPERTY_NAME = "DUMMY_USER_HOME";
+	private static final String MAVEN_LOCAL_REPO_PROPERTY_NAME = "maven.repo.local";
+	private static final String USER_HOME_PROPERTY_NAME = "user.home";
+
+	@Test
+	public void testUserSettingsWithEnvVariables() throws Exception {
+		final File localSettingsFile = new File(getClass().getResource("/settings-local-repo-test.xml").toURI());
+		String localSettingsContent = FileUtils.readFileToString(localSettingsFile, "UTF-8");
+		String localRepoProperty = System.getProperty(MAVEN_LOCAL_REPO_PROPERTY_NAME);
+		Properties properties = System.getProperties();
+		try {
+			properties.remove(MAVEN_LOCAL_REPO_PROPERTY_NAME); // Not to use local repository set from properties
+			
+			// Add a DUMMY environment variable to check later if it's really resolved to a real path
```

Comment:
It's a system property here, but the test is still useful.
As it's hard to change current env from inside Java, it could be interesting to write multiple tests:
* 1 for system property substitution (eg this one)
* another for env variable (here it could be any existing environment variable, let's just avoid $HOME which is used in default)

So it would test things separately and would bring a better coverage.

Reply:
There's not many env. variables we  can use... If not creating/setting the new ones,  at the first look we can use `MAVEN_PROJECTBASEDIR` which is set to the path to the current project (lemminx-maven in this case), or something like `PWD`, which is not a good one, imho, as it will make the test to create a `.m2/...` repository in current directory (which is a random thing that depends only on user that starts the script). I wouldn't add the cleaning/removal procedure for the repository here as, in theory, doing so we can remove the m2 repository that is normally used by other tests.

Reply:
Does the test create the repository? I was expecting that repository would be created lazily upon request, and here there is no request made to the repository. So if repo is not created, any env var can fit for testing (ie `${HOSTNAME}/.m2/bug383repository` and then comparing to `System.getEnv("HOSTNAME") + "/.m2/bug383repository"`

Reply:
Not sure if the repo is really created. For sure the test was creating it when I was invoking `initialize()` method of the Extension.
Now (as I said in other comment) - it is not created (and is not accessible as the result) But ptobably it's even better as we can use almost any env. variable to test it (and I just don't need to call `assertTrue(localRepositoryPath.canRead());` at the end of test case)

Reply:
The repositories aren't really created, probably, until the Maven Request is executed. We're only create the requests during the test. So no need to check the resultant  local repository for being accessible.